### ISSUE TITLE
dataset materialize for train and test data separately 

### ISSUE THREAD

**Issue by HoustonJ2013** [pyg-team/pytorch-frame#470]
The current example in the doc materialize the train and test before doing split. Could this lead to potential data leakage, as the col_stats is calculated on all data, but not training data. And for production test, should we have a separate function to materialise the data from raw data frame? 

from torch_frame.datasets import Yandex
from torch_frame.data import DataLoader

dataset = Yandex(root='/tmp/adult', name='adult')
dataset.materialize()
dataset.shuffle()
train_dataset, test_dataset = dataset[:0.8], dataset[0.80:]
train_loader = DataLoader(train_dataset.tensor_frame, batch_size=128,
                        shuffle=True)
test_loader = DataLoader(test_dataset.tensor_frame, batch_size=128)
